libmisc/logind.c

```diff
@@ -34,3 +34,18 @@ int get_session_host (char **out)
     free (session);
     return ret;
 }
+
+unsigned long active_sessions_count(const char *name, unsigned long unused)
+{
+    struct passwd *pw;
+    unsigned long count = 0;
+
+    pw = prefix_getpwnam(name);
+    if (pw == NULL) {
+        return 0;
+    }
+
+    count = sd_uid_get_sessions(pw->pw_uid, 0, NULL);
+
```

Comment:
Do you need to free pw?

Reply:
I'm not sure. Moreover, it is not clear to me that I have to use that interface and not another one (i.e. getpwnam) . I always get confused with these.

Reply:
Ok, yeah - you do not in fact want to free it :)  Sorry.